presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -219,23 +220,31 @@ private PlanBuilder appendScalarSubqueryApplyNode(PlanBuilder subPlan, SubqueryE
         subqueryPlan = subqueryPlan.withNewRoot(new EnforceSingleRowNode(idAllocator.getNextId(), subqueryPlan.getRoot()));
         subqueryPlan = subqueryPlan.appendProjections(coercions, symbolAllocator, idAllocator);
 
-        Assignments.Builder subqueryAssignments = Assignments.builder();
         Symbol uncoercedScalarSubquerySymbol = subqueryPlan.translate(uncoercedScalarSubquery);
         subPlan.getTranslations().put(uncoercedScalarSubquery, uncoercedScalarSubquerySymbol);
-        subqueryAssignments.put(uncoercedScalarSubquerySymbol, uncoercedScalarSubquerySymbol.toSymbolReference());
 
         for (Expression coercion : coercions) {
             Symbol coercionSymbol = subqueryPlan.translate(coercion);
             subPlan.getTranslations().put(coercion, coercionSymbol);
-            subqueryAssignments.put(coercionSymbol, coercionSymbol.toSymbolReference());
         }
 
-        return appendApplyNode(
-                subPlan,
-                scalarSubquery.getQuery(),
-                subqueryPlan,
-                subqueryAssignments.build(),
-                correlationAllowed);
+        PlanNode subqueryNode = subqueryPlan.getRoot();
```

Comment:
I think there is a way to make this code to be not repeated and to make this class be more readable, but I would prefer not to doing it as part of this PR.

I would extract another entity which would contain current processing state, like correlation, subquery node etc.